File: documentation/user/en/query/requirements/examples/hierarchy/hierarchy-from-node.java
final EvitaResponse<SealedEntity> result = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				// query hierarchy entity type
				collection("Product"),
				// target "Accessories" category
				filterBy(
					hierarchyWithin(
						"categories",
						attributeEquals("code", "audio")
					)
				),
				require(
					hierarchyOfReference(
						"categories",
						// request computation of direct children of the Portables category
						fromNode(
							"sideMenu1",
							node(
								filterBy(
									attributeEquals("code", "portables")
								)
							),
							entityFetch(attributeContent("code")),
							stopAt(distance(1)),
							statistics(
								CHILDREN_COUNT,
								QUERIED_ENTITY_COUNT
							)
						),
						// request computation of direct children of the Laptops category
						fromNode(
							"sideMenu2",
							node(
								filterBy(
									attributeEquals("code", "laptops")
								)
							),
							entityFetch(attributeContent("code")),
							stopAt(distance(1)),
							statistics(
								CHILDREN_COUNT,
								QUERIED_ENTITY_COUNT
							)
						)
					)
				)
			)
		);
	}
);

final Hierarchy hierarchyResult = result.getExtraResult(Hierarchy.class);
// index of both menu components
final Map<String, List<LevelInfo>> megaMenu = hierarchyResult.getReferenceHierarchy("categories");

File: documentation/user/en/query/requirements/examples/hierarchy/hierarchy-from-node.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("code", "audio")
        		)
        	),
        	Require(
        		HierarchyOfReference(
        			"categories",
        			RemoveEmpty,
        			FromNode(
        				"sideMenu1",
        				Node(
        					FilterBy(
        						AttributeEquals("code", "portables")
        					)
        				),
        				EntityFetch(
        					AttributeContent("code")
        				),
        				StopAt(
        					Distance(1)
        				),
        				Statistics(WithoutUserFilter, ChildrenCount, QueriedEntityCount)
        			),
        			FromNode(
        				"sideMenu2",
        				Node(
        					FilterBy(
        						AttributeEquals("code", "laptops")
        					)
        				),
        				EntityFetch(
        					AttributeContent("code")
        				),
        				StopAt(
        					Distance(1)
        				),
        				Statistics(WithoutUserFilter, ChildrenCount, QueriedEntityCount)
        			)
        		)
        	)
        )
	)
);